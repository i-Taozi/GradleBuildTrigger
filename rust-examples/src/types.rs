///类型
pub fn types() {
    let decimal = 65.4321_f32;
    //不支持隐式转化
    //let integer: u8 = decimal;
    // FIXME ^ Comment out this line

    //显示转化
    let integer = decimal as u8;
    let character = integer as char;

    println!("Casting: {} -> {} -> {}", decimal, integer, character);

    // 1000 already fits in a u16
    println!("1000 as a u16 is: {}", 1000 as u16);

    // 1000 - 256 - 256 - 256 = 232
    // Under the hood, the first 8 least significant bits (LSB) are kept,
    // while the rest towards the most significant bit (MSB) get truncated.
    //println!("1000 as a u8 is : {}", 1000 as u8);
    // -1 + 256 = 255
    println!("  -1 as a u8 is : {}", (-1i8) as u8);
    println!("1000 mod 256 is : {}", 1000 % 256);

    println!(" 128 as a i16 is: {}", 128 as i16);
    //println!(" 128 as a i8 is : {}", 128 as i8);
    //println!("1000 as a u8 is : {}", 1000 as u8);
    //println!(" 232 as a i8 is : {}", 232 as i8);

    literals();
}

fn literals() {
    //类型作为后缀
    let x = 1u8;
    let y = 2u32;
    let z = 3f32;

    //非固定文字，其类型取决于如何使用
    let i = 1;
    let f = 1.0;

    //Returns the size of the pointed-to value in bytes.
    println!("size of `x` in bytes: {}", std::mem::size_of_val(&x));
    println!("size of `y` in bytes: {}", std::mem::size_of_val(&y));
    println!("size of `z` in bytes: {}", std::mem::size_of_val(&z));
    println!("size of `i` in bytes: {}", std::mem::size_of_val(&i));
    println!("size of `f` in bytes: {}", std::mem::size_of_val(&f));
}
