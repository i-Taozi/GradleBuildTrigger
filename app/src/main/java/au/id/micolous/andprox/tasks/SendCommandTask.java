/*
 * This file is part of AndProx, an application for using Proxmark3 on Android.
 *
 * Copyright 2016-2019 Michael Farrell <micolous+git@gmail.com>
 *
 * This program is free software: you can redistribute it and/or modify it
 * under the terms of the GNU General Public License as published by the Free
 * Software Foundation, either version 3 of the License, or (at your option)
 * any later version.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
 * more details.
 *
 * You should have received a copy of the GNU General Public License along with
 * this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * Under section 7 of the GNU General Public License v3, the following additional
 * terms apply to this program:
 *
 *  (b) You must preserve reasonable legal notices and author attributions in
 *      the program.
 *  (c) You must not misrepresent the origin of this program, and need to mark
 *      modified versions in reasonable ways as different from the original
 *      version (such as changing the name and logos).
 *  (d) You may not use the names of licensors or authors for publicity
 *      purposes, without explicit written permission.
 */
package au.id.micolous.andprox.tasks;

import android.os.AsyncTask;

import java.lang.ref.WeakReference;

import au.id.micolous.andprox.natives.Natives;

/**
 * Dispatches commands to the PM3 client thread.
 */
public class SendCommandTask extends AsyncTask<String, Void, Void> {
    public interface SendCommandCallback {
        void onCommandFinished();
    }

    private static WeakReference<SendCommandCallback> callback = null;
    /**
     * This is a simple ref-counter for how many things are in flight.
     */
    private static int progressingCommands = 0;

    public static void register(SendCommandCallback c) {
        callback = new WeakReference<>(c);
    }

    /**
     * Returns the number of commands that are currently in progress.
     */
    public static int getProgressingCommands() {
        return progressingCommands;
    }

    private static SendCommandCallback getDoneCallback() {
        if (callback != null) {
            SendCommandCallback c = callback.get();
            if (c != null) {
                return c;
            }
        } else {
            // The weak reference has gone, also destroy it.
            callback = null;
        }
        return null;
    }

    @Override
    protected void onPreExecute() {
        super.onPreExecute();
        progressingCommands++;
    }

    @Override
    protected Void doInBackground(String... cmds) {
        for (String cmd : cmds) {
            Natives.sendCmd(cmd);
        }
        return null;
    }

    @Override
    protected void onPostExecute(Void aVoid) {
        progressingCommands--;
        SendCommandCallback c = getDoneCallback();
        if (c != null) c.onCommandFinished();
    }

    @Override
    protected void onCancelled() {
        super.onCancelled();

        progressingCommands--;
        SendCommandCallback c = getDoneCallback();
        if (c != null) c.onCommandFinished();
    }
}
