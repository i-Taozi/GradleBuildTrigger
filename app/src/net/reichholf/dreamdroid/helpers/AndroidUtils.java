package net.reichholf.dreamdroid.helpers;

import android.content.Context;
import android.content.pm.PackageManager;
import android.view.View;
import android.widget.TextView;

public class AndroidUtils {
	public static boolean isLeanback(Context ctx) {
		return (ctx.getPackageManager().hasSystemFeature(PackageManager.FEATURE_TELEVISION)
				|| ctx.getPackageManager().hasSystemFeature(PackageManager.FEATURE_LEANBACK));
	}

	public static void setTextOrHide(TextView textView, String text) {
		if (text.isEmpty())
			textView.setVisibility(View.GONE);
		else
			textView.setText(text);
	}
}
