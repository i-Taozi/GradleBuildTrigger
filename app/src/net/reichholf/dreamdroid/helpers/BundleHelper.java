/* Â© 2010 Stephan Reichholf <stephan at reichholf dot net>
 * 
 * Licensed under the Create-Commons Attribution-Noncommercial-Share Alike 3.0 Unported
 * http://creativecommons.org/licenses/by-nc-sa/3.0/
 */

package net.reichholf.dreamdroid.helpers;

import java.util.ArrayList;

/**
 * @author sre
 *
 */
public class BundleHelper {
	public static ArrayList<String> toStringArrayList(CharSequence[] strings){
		ArrayList<String> list = new ArrayList<>();
		for (CharSequence string : strings) {
			list.add(string.toString());
		}
		return list;
	}
	
	public static CharSequence[] toCharSequenceArray(ArrayList<String> strings){
		CharSequence[] list = new CharSequence[strings.size()];
		for(int i = 0; i < strings.size(); ++i){
			list[i] = strings.get(i);
		}
		return list;
	}
}
