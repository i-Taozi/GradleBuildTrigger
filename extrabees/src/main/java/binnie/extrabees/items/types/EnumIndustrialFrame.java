package binnie.extrabees.items.types;

import forestry.api.apiculture.IBeeGenome;
import forestry.api.apiculture.IBeeModifier;
import net.minecraft.item.Item;
import net.minecraft.item.ItemStack;
import net.minecraft.nbt.NBTTagCompound;

import javax.annotation.Nullable;

public enum EnumIndustrialFrame implements IBeeModifier {

	EMPTY("Empty", 5, 0),
	LIGHT("Glowstone Lighting", 2, 4) {
		@Override
		protected void init() {
			lighted = true;
		}
	},
	RAIN("Rain Shielding", 2, 4) {
		@Override
		protected void init() {
			rain = true;
		}
	},
	SUNLIGHT("Sunlight Simulator", 4, 8) {
		@Override
		protected void init() {
			lighted = true;
			sunlight = true;
		}
	},
	SOUL("Low Grade Mutagen", 5, 15) {
		@Override
		protected void init() {
			mutationMod = 1.3f;
		}
	},
	URANIUM("High Grade Mutagen", 10, 50) {
		@Override
		protected void init() {
			mutationMod = 2.0f;
		}
	},
	CAGE("Meshed Restrainer", 3, 12) {
		@Override
		protected void init() {
			territoryMod = 0.4f;
		}
	},
	FREEDOM("Territory Extension", 3, 16) {
		@Override
		protected void init() {
			territoryMod = 1.4f;
		}
	},
	HONEY("Honey Amplifier", 4, 12) {
		@Override
		protected void init() {
			productionMod = 1.4f;
		}
	},
	JELLY("Gelatin Amplifier", 8, 36) {
		@Override
		protected void init() {
			productionMod = 1.8f;
		}
	},
	LEAF("Pollinator MK I", 3, 15) {
		@Override
		protected void init() {
			floweringMod = 1.4f;
		}
	},
	POLLEN("Pollinator MK II", 7, 25) {
		@Override
		protected void init() {
			floweringMod = 2.0f;
		}
	},
	CLAY("Lifespan Extensor", 2, 10) {
		@Override
		protected void init() {
			lifespanMod = 1.4f;
		}
	},
	EMERALD("Eon Simulator", 7, 20) {
		@Override
		protected void init() {
			lifespanMod = 2.0f;
		}
	},
	NETHER_STAR("Immortality Gate", 12, 50) {
		@Override
		protected void init() {
			lifespanMod = 20.0f;
		}
	},
	POISON("Mortality Inhibitor", 8, 18) {
		@Override
		protected void init() {
			lifespanMod = 0.5f;
		}
	};

	private final String name;
	protected float territoryMod;
	protected float mutationMod;
	protected float lifespanMod;
	protected float productionMod;
	protected float floweringMod;
	protected boolean lighted;
	protected boolean sunlight;
	protected boolean rain;
	private final int wearMod;
	private final int power;

	EnumIndustrialFrame(final String name, final int wear, final int power) {
		this.territoryMod = 1.0f;
		this.mutationMod = 1.0f;
		this.lifespanMod = 1.0f;
		this.productionMod = 1.0f;
		this.floweringMod = 1.0f;
		this.lighted = false;
		this.sunlight = false;
		this.rain = false;
		this.name = name;
		this.wearMod = wear;
		this.power = power;
		init();
	}

	protected void init() {

	}

	public static ItemStack getItemStack(final Item item, final EnumIndustrialFrame frame) {
		final ItemStack stack = new ItemStack(item);
		final NBTTagCompound nbt = new NBTTagCompound();
		nbt.setInteger("frame", frame.ordinal());
		stack.setTagCompound(nbt);
		return stack;
	}

	@Override
	public float getTerritoryModifier(final IBeeGenome genome, final float currentModifier) {
		return this.territoryMod;
	}

	@Override
	public float getMutationModifier(final IBeeGenome genome, final IBeeGenome mate, final float currentModifier) {
		return this.mutationMod;
	}

	@Override
	public float getLifespanModifier(final IBeeGenome genome, @Nullable final IBeeGenome mate, final float currentModifier) {
		return this.lifespanMod;
	}

	@Override
	public float getProductionModifier(final IBeeGenome genome, final float currentModifier) {
		return this.productionMod;
	}

	@Override
	public float getFloweringModifier(final IBeeGenome genome, final float currentModifier) {
		return this.floweringMod;
	}

	@Override
	public boolean isSealed() {
		return this.rain;
	}

	@Override
	public boolean isSelfLighted() {
		return this.lighted;
	}

	@Override
	public boolean isSunlightSimulated() {
		return this.sunlight;
	}

	@Override
	public boolean isHellish() {
		return false;
	}

	public String getName() {
		return this.name;
	}

	public int getWearModifier() {
		return this.wearMod;
	}

	public int getPowerUsage() {
		return this.power;
	}

	@Override
	public float getGeneticDecay(final IBeeGenome genome, final float currentModifier) {
		return 1.0f;
	}

}
