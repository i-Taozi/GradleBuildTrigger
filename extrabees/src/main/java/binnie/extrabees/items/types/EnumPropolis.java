package binnie.extrabees.items.types;

import binnie.core.util.I18N;
import binnie.extrabees.modules.ModuleCore;
import binnie.extrabees.utils.Utils;
import forestry.api.recipes.RecipeManagers;
import net.minecraft.item.ItemStack;
import net.minecraftforge.fluids.FluidStack;

public enum EnumPropolis implements IEBEnumItem {
	WATER(2405321, 12762791, "Water"),
	OIL(1519411, 12762791, "oil"),
	FUEL(10718482, 12762791, "fuel"),
	MILK,
	FRUIT,
	SEED,
	ALCOHOL,
	CREOSOTE(8877313, 12428819, "creosote"),
	GLACIAL,
	PEAT;

	private final int primaryColor;
	private final int secondaryColor;
	private final String liquidName;
	private boolean active;

	EnumPropolis() {
		this(16777215, 16777215, "");
		this.active = false;
	}

	EnumPropolis(int primaryColor, int secondaryColor, String liquidName) {
		this.active = true;
		this.primaryColor = primaryColor;
		this.secondaryColor = secondaryColor;
		this.liquidName = liquidName;
	}

	public static EnumPropolis get(final ItemStack itemStack) {
		final int i = itemStack.getItemDamage();
		if (i >= 0 && i < values().length) {
			return values()[i];
		}
		return values()[0];
	}

	public void addRecipe() {
		final FluidStack liquid = Utils.getFluidFromName(this.liquidName, 500);
		if (liquid != null) {
			RecipeManagers.squeezerManager.addRecipe(20, this.get(1), liquid, ItemStack.EMPTY, 0);
		}
	}

	public int getSpriteColour(int renderPass) {
		if (renderPass == 0) {
			return primaryColor;
		}
		if (renderPass == 1) {
			return secondaryColor;
		}
		return 0xffffff;
	}

	@Override
	public boolean isActive() {
		return this.active && Utils.getFluidFromName(this.liquidName, 100) != null;
	}

	@Override
	public ItemStack get(final int amount) {
		return new ItemStack(ModuleCore.propolis, amount, this.ordinal());
	}

	@Override
	public String getName(final ItemStack itemStack) {
		return I18N.localise("extrabees.item.propolis." + this.name().toLowerCase());
	}

}
