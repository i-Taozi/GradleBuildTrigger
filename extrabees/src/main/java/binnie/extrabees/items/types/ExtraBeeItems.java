package binnie.extrabees.items.types;

import binnie.extrabees.modules.ModuleCore;
import net.minecraft.item.ItemStack;
import net.minecraft.util.NonNullList;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;
import net.minecraftforge.oredict.OreDictionary;

import javax.annotation.Nullable;
import java.util.List;

public enum ExtraBeeItems implements IEBItemMiscProvider {

	SCENTED_GEAR("Scented Gear", "scented_gear"),
	DIAMOND_SHARD("Diamond Fragment", "diamond_shard") {
		@Override
		protected void init() {
			setGem("Diamond");
		}
	},
	EMERALD_SHARD("Emerald Fragment", "emerald_shard") {
		@Override
		protected void init() {
			setGem("Emerald");
		}
	},
	RUBY_SHARD("Ruby Fragment", "ruby_shard") {
		@Override
		protected void init() {
			setGem("Ruby");
		}
	},
	SAPPHIRE_SHARD("Sapphire Fragment", "sapphire_shard") {
		@Override
		protected void init() {
			setGem("Sapphire");
		}
	},
	LAPIS_SHARD("Lapis Fragment", "lapis_shard"),
	IRON_DUST("Iron Grains", "iron_dust") {
		@Override
		protected void init() {
			setMetal("Iron");
		}
	},
	GOLD_DUST("Gold Grains", "gold_dust") {
		@Override
		protected void init() {
			setMetal("Gold");
		}
	},
	SILVER_DUST("Silver Grains", "silver_dust") {
		@Override
		protected void init() {
			setMetal("Silver");
		}
	},
	PLATINUM_DUST("Platinum Grains", "platinum_dust") {
		@Override
		protected void init() {
			setMetal("Platinum");
		}
	},
	COPPER_DUST("Copper Grains", "copper_dust") {
		@Override
		protected void init() {
			setMetal("Copper");
		}
	},
	TIN_DUST("Tin Grains", "tin_dust") {
		@Override
		protected void init() {
			setMetal("Tin");
		}
	},
	NICKEL_DUST("Nickel Grains", "nickel_dust") {
		@Override
		protected void init() {
			setMetal("Nickel");
		}
	},
	LEAD_DUST("Lead Grains", "lead_dust") {
		@Override
		protected void init() {
			setMetal("Lead");
		}
	},
	ZINC_DUST("Zinc Grains", "zinc_dust") {
		@Override
		protected void init() {
			setMetal("Zinc");
		}
	},
	TITANIUM_DUST("Titanium Grains", "titanium_dust") {
		@Override
		protected void init() {
			setMetal("Titanium");
		}
	},
	TUNGSTEN_DUST("Tungsten Grains", "tungsten_dust") {
		@Override
		protected void init() {
			setMetal("Tungsten");
		}
	},
	URANIUM_DUST("Radioactive Fragments", "radioactive_dust"),
	COAL_DUST("Coal Grains", "coal_dust") {
		@Override
		protected void init() {
			setMetal("Coal");
		}
	},
	RED_DYE("Red Dye", "dye_red"),
	YELLOW_DYE("Yellow Dye", "dye_yellow"),
	BLUE_DYE("Blue Dye", "dye_blue"),
	GREEN_DYE("Green Dye", "dye_green"),
	WHITE_DYE("White Dye", "dye_white"),
	BLACK_DYE("Black Dye", "dye_black"),
	BROWN_DYE("Brown Dye", "dye_brown"),
	CLAY_DUST("Clay Dust", "clay_dust"),
	YELLORIUM_DUST("Yellorium Grains", "yellorium_dust") {
		@Override
		protected void init() {
			setMetal("Yellorium");
		}
	},
	BLUTONIUM_DUST("Blutonium Grains", "blutonium_dust") {
		@Override
		protected void init() {
			setMetal("Blutonium");
		}
	},
	CYANITE_DUST("Cyanite Grains", "cyanite_dust") {
		@Override
		protected void init() {
			setMetal("Cyanite");
		}
	};

	public final String name;
	public final String modelPath;
	@Nullable
	public String metalString;
	@Nullable
	public String gemString;

	ExtraBeeItems(String name, String modelPath) {
		this.metalString = null;
		this.gemString = null;
		this.name = name;
		this.modelPath = modelPath;
		init();
	}

	protected void setGem(final String string) {
		this.gemString = string;
	}

	protected void setMetal(final String string) {
		this.metalString = string;
	}

	@Override
	public boolean isActive() {
		if (this.metalString != null) {
			NonNullList<ItemStack> ingots = OreDictionary.getOres("ingot" + this.metalString);
			NonNullList<ItemStack> dust = OreDictionary.getOres("dust" + this.metalString);
			return !ingots.isEmpty() || !dust.isEmpty() || this == ExtraBeeItems.COAL_DUST;
		}
		NonNullList<ItemStack> gems = OreDictionary.getOres("gem" + this.gemString);
		return this.gemString == null || !gems.isEmpty();
	}

	protected void init() {
	}

	@Override
	@SideOnly(Side.CLIENT)
	public void addInformation(final List<String> tooltip) {
	}

	@Override
	public String getName(final ItemStack itemStack) {
		return this.name;
	}

	@Override
	public ItemStack get(final int amount) {
		return new ItemStack(ModuleCore.itemMisc, amount, this.ordinal());
	}

	@Override
	public String getModelPath() {
		return modelPath;
	}

}
