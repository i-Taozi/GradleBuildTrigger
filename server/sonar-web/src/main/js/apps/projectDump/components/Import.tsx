/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import classNames from 'classnames';
import { stringify } from 'querystring';
import * as React from 'react';
import { doImport } from '../../../api/project-dump';
import { Button } from '../../../components/controls/buttons';
import DateFromNow from '../../../components/intl/DateFromNow';
import DateTimeFormatter from '../../../components/intl/DateTimeFormatter';
import { Alert } from '../../../components/ui/Alert';
import { translate, translateWithParameters } from '../../../helpers/l10n';
import { getBaseUrl } from '../../../helpers/system';
import { DumpStatus, DumpTask } from '../../../types/project-dump';
import { TaskStatuses, TaskTypes } from '../../../types/tasks';

interface Props {
  analysis?: DumpTask;
  componentKey: string;
  importEnabled: boolean;
  loadStatus: () => void;
  status: DumpStatus;
  task?: DumpTask;
}

export default class Import extends React.Component<Props> {
  handleImport = () => {
    doImport(this.props.componentKey).then(this.props.loadStatus, () => {
      /* no catch needed */
    });
  };

  renderWhenCanNotImport() {
    return (
      <div className="boxed-group-inner" id="import-not-possible">
        {translate('project_dump.can_not_import')}
      </div>
    );
  }

  renderWhenNoDump() {
    return (
      <div className="boxed-group-inner">
        <Alert id="import-no-file" variant="warning">
          {translate('project_dump.no_file_to_import')}
        </Alert>
      </div>
    );
  }

  renderImportForm() {
    return (
      <div>
        <div className="spacer-bottom">{translate('project_dump.import_form_description')}</div>
        <Button onClick={this.handleImport}>{translate('project_dump.do_import')}</Button>
      </div>
    );
  }

  renderWhenImportSuccess(task: DumpTask) {
    return (
      <div className="boxed-group-inner">
        {task.executedAt && (
          <DateTimeFormatter date={task.executedAt}>
            {formatted => (
              <Alert variant="success">
                {translateWithParameters('project_dump.import_success', formatted)}
              </Alert>
            )}
          </DateTimeFormatter>
        )}
      </div>
    );
  }

  renderWhenImportPending(task: DumpTask) {
    return (
      <div className="boxed-group-inner" id="import-pending">
        <i className="spinner spacer-right" />
        <DateTimeFormatter date={task.submittedAt}>
          {formatted => (
            <span>{translateWithParameters('project_dump.pending_import', formatted)}</span>
          )}
        </DateTimeFormatter>
      </div>
    );
  }

  renderWhenImportInProgress(task: DumpTask) {
    return (
      <div className="boxed-group-inner" id="import-in-progress">
        <i className="spinner spacer-right" />
        {task.startedAt && (
          <DateFromNow date={task.startedAt}>
            {fromNow => (
              <span>{translateWithParameters('project_dump.in_progress_import', fromNow)}</span>
            )}
          </DateFromNow>
        )}
      </div>
    );
  }

  renderWhenImportFailed() {
    const { componentKey } = this.props;
    const detailsUrl = `${getBaseUrl()}/project/background_tasks?${stringify({
      id: encodeURIComponent(componentKey),
      status: TaskStatuses.Failed,
      taskType: TaskTypes.ProjectImport
    })}`;

    return (
      <div className="boxed-group-inner">
        <Alert id="export-in-progress" variant="error">
          {translate('project_dump.failed_import')}
          <a className="spacer-left" href={detailsUrl}>
            {translate('project_dump.see_details')}
          </a>
        </Alert>

        {this.renderImportForm()}
      </div>
    );
  }

  render() {
    const { importEnabled, status, task, analysis } = this.props;

    let content: React.ReactNode = null;
    if (task && task.status === TaskStatuses.Success && !analysis) {
      content = this.renderWhenImportSuccess(task);
    } else if (task && task.status === TaskStatuses.Pending) {
      content = this.renderWhenImportPending(task);
    } else if (task && task.status === TaskStatuses.InProgress) {
      content = this.renderWhenImportInProgress(task);
    } else if (task && task.status === TaskStatuses.Failed) {
      content = this.renderWhenImportFailed();
    } else if (!status.canBeImported) {
      content = this.renderWhenCanNotImport();
    } else if (!status.dumpToImport) {
      content = this.renderWhenNoDump();
    } else {
      content = <div className="boxed-group-inner">{this.renderImportForm()}</div>;
    }
    return (
      <div
        className={classNames('boxed-group', {
          'import-disabled text-muted': !importEnabled
        })}
        id="project-import">
        <div className="boxed-group-header">
          <h2>{translate('project_dump.import')}</h2>
        </div>
        {importEnabled ? (
          content
        ) : (
          <div className="boxed-group-inner">
            {translate('project_dump.import_form_description_disabled')}
          </div>
        )}
      </div>
    );
  }
}
