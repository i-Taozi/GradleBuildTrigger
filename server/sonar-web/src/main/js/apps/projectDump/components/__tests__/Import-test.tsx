/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { shallow } from 'enzyme';
import * as React from 'react';
import { mockDumpStatus, mockDumpTask } from '../../../../helpers/testMocks';
import { TaskStatuses } from '../../../../types/tasks';
import Import from '../Import';

jest.mock('../../../../api/project-dump', () => ({
  doImport: jest.fn().mockResolvedValue({})
}));

it('should render correctly', () => {
  expect(
    shallowRender({ status: mockDumpStatus({ dumpToImport: 'import-file.zip' }) })
  ).toMatchSnapshot('import form');
  expect(shallowRender()).toMatchSnapshot('no dump to import');
  expect(shallowRender({ status: mockDumpStatus({ canBeImported: false }) })).toMatchSnapshot(
    'cannot import'
  );
  expect(shallowRender({ task: mockDumpTask({ status: TaskStatuses.Success }) })).toMatchSnapshot(
    'success'
  );
  expect(
    shallowRender({
      analysis: mockDumpTask(),
      task: mockDumpTask({ status: TaskStatuses.Success })
    })
  ).toMatchSnapshot('success, but with analysis -> show form');
  expect(shallowRender({ task: mockDumpTask({ status: TaskStatuses.Pending }) })).toMatchSnapshot(
    'pending'
  );
  expect(
    shallowRender({ task: mockDumpTask({ status: TaskStatuses.InProgress }) })
  ).toMatchSnapshot('in progress');
  expect(shallowRender({ task: mockDumpTask({ status: TaskStatuses.Failed }) })).toMatchSnapshot(
    'failed'
  );
  expect(shallowRender({ importEnabled: false })).toMatchSnapshot('import disabled');
});

function shallowRender(overrides: Partial<Import['props']> = {}) {
  return shallow<Import>(
    <Import
      importEnabled={true}
      analysis={undefined}
      componentKey="key"
      loadStatus={jest.fn()}
      status={mockDumpStatus()}
      task={undefined}
      {...overrides}
    />
  );
}
