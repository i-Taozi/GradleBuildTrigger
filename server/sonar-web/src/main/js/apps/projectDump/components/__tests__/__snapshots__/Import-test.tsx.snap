// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly: cannot import 1`] = `
<div
  className="boxed-group"
  id="project-import"
>
  <div
    className="boxed-group-header"
  >
    <h2>
      project_dump.import
    </h2>
  </div>
  <div
    className="boxed-group-inner"
    id="import-not-possible"
  >
    project_dump.can_not_import
  </div>
</div>
`;

exports[`should render correctly: failed 1`] = `
<div
  className="boxed-group"
  id="project-import"
>
  <div
    className="boxed-group-header"
  >
    <h2>
      project_dump.import
    </h2>
  </div>
  <div
    className="boxed-group-inner"
  >
    <Alert
      id="export-in-progress"
      variant="error"
    >
      project_dump.failed_import
      <a
        className="spacer-left"
        href="/project/background_tasks?id=key&status=FAILED&taskType=PROJECT_IMPORT"
      >
        project_dump.see_details
      </a>
    </Alert>
    <div>
      <div
        className="spacer-bottom"
      >
        project_dump.import_form_description
      </div>
      <Button
        onClick={[Function]}
      >
        project_dump.do_import
      </Button>
    </div>
  </div>
</div>
`;

exports[`should render correctly: import disabled 1`] = `
<div
  className="boxed-group import-disabled text-muted"
  id="project-import"
>
  <div
    className="boxed-group-header"
  >
    <h2>
      project_dump.import
    </h2>
  </div>
  <div
    className="boxed-group-inner"
  >
    project_dump.import_form_description_disabled
  </div>
</div>
`;

exports[`should render correctly: import form 1`] = `
<div
  className="boxed-group"
  id="project-import"
>
  <div
    className="boxed-group-header"
  >
    <h2>
      project_dump.import
    </h2>
  </div>
  <div
    className="boxed-group-inner"
  >
    <div>
      <div
        className="spacer-bottom"
      >
        project_dump.import_form_description
      </div>
      <Button
        onClick={[Function]}
      >
        project_dump.do_import
      </Button>
    </div>
  </div>
</div>
`;

exports[`should render correctly: in progress 1`] = `
<div
  className="boxed-group"
  id="project-import"
>
  <div
    className="boxed-group-header"
  >
    <h2>
      project_dump.import
    </h2>
  </div>
  <div
    className="boxed-group-inner"
    id="import-in-progress"
  >
    <i
      className="spinner spacer-right"
    />
    <DateFromNow
      date="2020-03-12T12:20:20Z"
    >
      <Component />
    </DateFromNow>
  </div>
</div>
`;

exports[`should render correctly: no dump to import 1`] = `
<div
  className="boxed-group"
  id="project-import"
>
  <div
    className="boxed-group-header"
  >
    <h2>
      project_dump.import
    </h2>
  </div>
  <div
    className="boxed-group-inner"
  >
    <Alert
      id="import-no-file"
      variant="warning"
    >
      project_dump.no_file_to_import
    </Alert>
  </div>
</div>
`;

exports[`should render correctly: pending 1`] = `
<div
  className="boxed-group"
  id="project-import"
>
  <div
    className="boxed-group-header"
  >
    <h2>
      project_dump.import
    </h2>
  </div>
  <div
    className="boxed-group-inner"
    id="import-pending"
  >
    <i
      className="spinner spacer-right"
    />
    <DateTimeFormatter
      date="2020-03-12T12:15:20Z"
    >
      <Component />
    </DateTimeFormatter>
  </div>
</div>
`;

exports[`should render correctly: success 1`] = `
<div
  className="boxed-group"
  id="project-import"
>
  <div
    className="boxed-group-header"
  >
    <h2>
      project_dump.import
    </h2>
  </div>
  <div
    className="boxed-group-inner"
  >
    <DateTimeFormatter
      date="2020-03-12T12:22:20Z"
    >
      <Component />
    </DateTimeFormatter>
  </div>
</div>
`;

exports[`should render correctly: success, but with analysis -> show form 1`] = `
<div
  className="boxed-group"
  id="project-import"
>
  <div
    className="boxed-group-header"
  >
    <h2>
      project_dump.import
    </h2>
  </div>
  <div
    className="boxed-group-inner"
  >
    <Alert
      id="import-no-file"
      variant="warning"
    >
      project_dump.no_file_to_import
    </Alert>
  </div>
</div>
`;
